### DISCUSSION TITLE
Q: vcpkg (VS2022 vs standalone) and building pybind11 port

### DISCUSSION THREAD

**Discussion by misuo** [microsoft/vcpkg#33391] [Q&A] [answered]
I'm trying to use the **vcpkg 2023-06-22** coming with **Visual Studio 2022 17.7** in manifest mode.

My vcpkg.json file contains reference to pybind11 port but after after doing `vcpkg install` it doesn't build the pybind11 port (or any dependencies), only download. 

On another machine using standalone **vcpkg 2022-09-01** everything works (both download and fully build with dependencies). Q1: Anybody can explain this?

Q2: Is there a difference in this regard between the version coming with Visual Studio 2022 and standalone vcpkg?

Q3: Why is it that an older version of vcpkg uses (downloads) older versions of ports? E.g. the **vcpkg 2022-09-01** downloads pybind11 v2.10.0#1 where as **vcpkg 2023-06-22** downloads pybind11 v2.11.1. 

The missing build may be pybind11 port related rather than vcpkg related, I don't think I explicitly can trigger a build of a port through vcpkg.

Thanks in advance for any help/advice.

PS. Using the **vcpkg 2023-06-22** seems to require baseline so I've executed the `vcpkg x-update-baseline --add-initial-baseline` to get that. I don't really understand what this is or whether that is causing the problem.

**Comment by Osyotr** [comment, marked as answer]
> My vcpkg.json file contains reference to pybind11 port but after after doing vcpkg install it doesn't build the pybind11 port 

pybind11 is a header-only library, there's nothing to build. Are you missing python3 dependency in you manifest?

**Reply by misuo** [reply]
Well I didn't need to have python3 listed in the vcpkg.json in the past. So this must be something new. Thx for the suggestion.

I believe this might be the right answer, Let me check...

**Reply by Osyotr** [reply]
Yeah, it was removed somewhere between 2022-09-01 and 2023-06-22. The reason might be that pybind11 supported python2 back then (now it does not).
